### DISCUSSION TITLE
cases

### DISCUSSION THREAD

**Discussion by APT256** [whatotter/pwnhyve#13] [General]
So I am no good with CAD and I was hoping that someone in the group might have designed an enclosure for the Pi Zero and display? If so, would you be willing to share the stl file?

**Comment by whatotter** [comment]
i haven't been working on one yet, since it depends on your setup, but maybe after i get RF and maybe even NFC working, i'll start working on a basic one

let me know your setup, and i'll work on it and send it here

**Reply by APT256** [reply]
@whatotter thanks for the offer. By all means don't take time away from the important stuff for this. As far as my setup goes, it is a Pi Zero W v2 with a 1.3inch OLED Display HAT.

**Reply by whatotter** [reply]
will work on it eventually - my pi zero's screen cracked and unfortunately doesn't work (no clue how that happened), so i can only work on some basic stuff in the meanwhile

**Reply by APT256** [reply]
Bummer! If you want to send me a shipping address, I'll have a new one drop-shipped to you. The least I can do for all the work you have put into this!

**Reply by whatotter** [reply]
nah ill be fine - i already ordered a new one! it should come in about next week, so i just gotta be patient
thanks for the offer though, much appreciated 🙏

**Reply by APT256** [reply]
If there is anything I can do to help, just let me know.

**Reply by whatotter** [reply]
will do 🙏

**Reply by APT256** [reply]
@whatotter just curious if you have had a chance to work on a case?

**Reply by whatotter** [reply]
yes i have - i'm mimicking the flipper zero's design (since it also protects the screen, and provides a lot of space (i'm thinking of designing another one, smaller)) and it's been going somewhat well  

i've been stuck on finding clips that are able to be 3d printed and still strong, but i haven't found any and screws aren't really an option

![image](https://github.com/whatotter/pwnhyve/assets/42103041/f3c910b1-6939-4ab6-9301-0d105a16a6f6)

i can send stls for you to try, but i'm not 100% certain they'll work

**Reply by APT256** [reply]
Looks good, I'll get my son to print it for me and let you know how it works.

**Reply by whatotter** [reply]
just a quick reminder 👍 i believe the sd card might not fit in the slot i made for the pi, but it also might, so its 50/50

**Reply by APT256** [reply]
@whatotter I have the replacement parts for my printer on order. Would you mind sending me you stl files for the case and I will use them as a starting point.

**Comment by APT256** [comment]
Sure, I’ll give it a shot and if i come up with any improvements I’ll let you know.v/rJonOn Nov 26, 2023, at 18:22, otter ***@***.***> wrote:﻿
yes i have - i'm mimicking the flipper zero's design (since it also protects the screen, and provides a lot of space (i'm thinking of designing another one, smaller)) and it's been going somewhat well
i've been stuck on finding clips that are able to be 3d printed and still strong, but i haven't found any and screws aren't really an option
image.png (view on web)
i can send stls for you to try, but i'm not 100% certain they'll work

**Reply by whatotter** [reply]
supports shouldn't be needed, but its up to you

[flipper.zip](https://github.com/whatotter/pwnhyve/files/13468826/flipper.zip)

**Reply by APT256** [reply]
roger...

**Reply by APT256** [reply]
So I finally bought myself a 3d printer (Creality K1C). I am now comfortable enough with it, that I will go ahead print your design. If there are issues with access to the SDIO card, I will ask my wife to mod the design (she is a mechanical engineer and is pretty good with CAD). I will post updates here. If you would rather I put them somewhere else, just let me know.

**Reply by whatotter** [reply]
here or your own discussion, it's up to you :)

i do think that the case _should_ fit, but i do plan on doing a redesign to take in account the PCB shim prototype, aswell as a battery and having some padding for the screen (the waveshare OLED display itself is quite weak).

i too am not that good at CAD, so having your wife help both of us with this would be a blessing 😁

**Reply by APT256** [reply]
Well, crud, my printer just had a major blow-out at the hot end. Looks like I am going to have to order some replacement parts before I can take a look at this.

**Comment by APT256** [comment]
So i just got the USB hat in and installed. It isn't as bulky as I was worried about. Now, if I can get my 3d printer working again, I will work on getting a case designed. Any chance that I can get my hands on a shim so I can work that into the design?

![image](https://github.com/user-attachments/assets/6d2fd436-c771-4f86-9a36-c07921f58d39)

**Reply by whatotter** [reply]
looks extremely cool, i will add the PCB gerber files for the shim to the repo in a bit so you can order a couple boards with your favorite PCB fabricator (i use jlcpcb)

sidenote, it does come with a spot to add a charging module so you can have a slimmer rpi with a bigger battery

**Reply by APT256** [reply]
Cool!

**Comment by igeometrix** [comment]
I've got a bambu labs X1 Carbon if you guys need any parts I'd be happy to
help. I would love to get a version for myself!

**Reply by whatotter** [reply]
will do, i lost the original stl files (not like they were going to fit anyways... tolerances) so I'd have to redo them - i just finished working on my vzbot so once i get some prototypes finished you and @servebeer will be the first with the cases 🎉

**Reply by APT256** [reply]
@igeometrix how are you with CAD? I am not good and dependent on my wife for design work. If you are willing to do the design work, I am more than happy to help where I can. If not, no problem, it might be a little slow going depending on her work load.

**Comment by APT256** [comment]
Thanks a lot whatotter! BTW, I was looking for the design files for the shim but could not locate them. I am sure I am looking in the wrong place. Would you mind sending them to me, or posting the URL to their location?

v/r
Jon
